src/parser/jobs/gnb/modules/NoMercy.tsx

```diff
@@ -37,228 +34,87 @@ const EXPECTED_USES = {
 	// Don't check for correctness on the Gnashing Fang combo; that's covered by the built-in Combo tracker.
 }
 
-const NO_MERCY_BUFF_DURATION = 20000 // in milliseconds
-
-class NoMercyState {
-	start: number
-	end?: number
-	rotation: CastEvent[] = []
-	isRushing: boolean = false
-
-	// Track these for pre-processing so we don't have to loop back over this
-	// a bunch of times later.
-	numGcds: number = 0
-	numBlastingZones: number = 0
-	numSonicBreaks: number = 0
-	numRoughDivides: number = 0
-	numGnashingFangs: number = 0
-	numBowShocks: number = 0
-
-	constructor(start: number) {
-		this.start = start
-	}
-}
-
-export default class NoMercy extends Module {
+export default class NoMercy extends BuffWindowModule {
 	static handle = 'nomercy'
 	static title = t('gnb.nomercy.title')`No Mercy Windows`
 
-	@dependency private suggestions!: Suggestions
-	@dependency private timeline!: Timeline
+	buffAction = ACTIONS.NO_MERCY
+	buffStatus = STATUSES.NO_MERCY
```

Comment:
Should these use the data system?

Reply:
the other implementations of BuffWindow don't use data, so I didn't.

Is there a reason to use data over just using actions/statuses directly here?

Reply:
You just want the ID so not really. The main benefit of the data system is it more properly manages layers for patch changes, but you literally just want the ID here anyway as far as I can tell.